app/src/pages/eligibility.tsx

```diff
@@ -1,4 +1,3 @@
-import { useAppContext } from '@context/state'
```

Comment:
Notes for reviewers: The Eligibility page (`/eligibility`) is the first editable form page in the form wizard flow. The back link routes to `/how-it-works` and the call to action button routes the user to either `/other-benefits`, `/income`, or `/choose-clinic` depending on how they answer the questions on this page (This is just for reviewer context; that logic is not handled in this page). There are no data validation guards for this page because it is the first editable page in the form wizard flow.

This form page establishes a pattern that is used in all the remaining form pages. At a high level:
- User data is stored first on a per-page basis in a state called `form`
- This state is initialized with blank data from `/src/utils/sessionData.ts` (added in #142) rather than the `session` prop in order to avoid _really weird_ react state bugs
- Using a `useEffect()`, anytime `form` is updated, we also update the global state called `session` which is owned by `_app.tsx` and passed in as a prop 
- (Not shown in this commit) `session` is both a global state and is stored in the browser's session storage to enable the user to refresh the page without losing state
- We have a state called `disabled` that manages whether the call to action button at the bottom of the page is enabled/disabled. It should only be enabled when all required fields have been filled out
- We have a `handleChange` function that is passed to each form element. Whenever the user changes a form element, the local `form` state and the global `session` state are updated

![Screen Shot 2022-10-28 at 1 51 55 AM](https://user-images.githubusercontent.com/67701/198546628-ad98fbe1-27c1-454b-a96a-f638ec9ff205.png)

Reply:
Note to reviewers: I don't think there's time left on this project to modify this high level pattern right now. However, it would be really helpful to know if there is a better or more established pattern for accomplishing the same thing.

Reply:
@sawyerh Thank you!

- 🤔 That's a good call. I forgot about the `??` nullish coalescing operator, which I used on CA UI. I wonder if I have any cases where the initial form state is necessary in some way, but it would be really nice if I was able to toss that whole chunk of logic
- I hadn't come across `react-hook-form` yet, though I have seen and was thinking about [`formik`](https://formik.org/). It's neat however, to see that starting in [May 2022](https://npmtrends.com/formik-vs-react-hook-form-vs-redux-form), `react-hook-form` has become more popular than `formik`.